x-pack/plugins/cases/public/common/lib/kibana/hooks.ts

```diff
@@ -0,0 +1,134 @@
+
+import { DEFAULT_DATE_FORMAT, DEFAULT_DATE_FORMAT_TZ } from '../../../../common/constants';
+import { errorToToaster, useStateToaster } from '../../../components/toasters';
+import { AuthenticatedUser } from '../../../../../security/common/model';
+import { convertToCamelCase } from '../../../containers/utils';
+import { StartServices } from '../../../types';
+import { useUiSetting, useKibana } from './kibana_react';
+
+export const useDateFormat = (): string => useUiSetting<string>(DEFAULT_DATE_FORMAT);
+
+export const useTimeZone = (): string => {
+  const timeZone = useUiSetting<string>(DEFAULT_DATE_FORMAT_TZ);
+  return timeZone === 'Browser' ? moment.tz.guess() : timeZone;
+};
+
+export const useBasePath = (): string => useKibana().services.http.basePath.get();
+
+export const useToasts = (): StartServices['notifications']['toasts'] =>
+  useKibana().services.notifications.toasts;
+
+export const useHttp = (): StartServices['http'] => useKibana().services.http;
+
+interface UserRealm {
+  name: string;
+  type: string;
+}
+
+export interface AuthenticatedElasticUser {
+  username: string;
+  email: string;
+  fullName: string;
+  roles: string[];
+  enabled: boolean;
+  metadata?: {
+    _reserved: boolean;
+  };
+  authenticationRealm: UserRealm;
+  lookupRealm: UserRealm;
+  authenticationProvider: string;
+}
+
+export const useCurrentUser = (): AuthenticatedElasticUser | null => {
+  const [user, setUser] = useState<AuthenticatedElasticUser | null>(null);
+
+  const [, dispatchToaster] = useStateToaster();
+
+  const { security } = useKibana().services;
+
+  const fetchUser = useCallback(() => {
+    let didCancel = false;
+    const fetchData = async () => {
+      try {
+        if (security != null) {
+          const response = await security.authc.getCurrentUser();
+          if (!didCancel) {
+            setUser(convertToCamelCase<AuthenticatedUser, AuthenticatedElasticUser>(response));
+          }
+        } else {
+          setUser({
+            username: i18n.translate('xpack.cases.getCurrentUser.unknownUser', {
+              defaultMessage: 'Unknown',
+            }),
+            email: '',
+            fullName: '',
+            roles: [],
+            enabled: false,
+            authenticationRealm: { name: '', type: '' },
+            lookupRealm: { name: '', type: '' },
+            authenticationProvider: '',
+          });
+        }
+      } catch (error) {
+        if (!didCancel) {
+          errorToToaster({
+            title: i18n.translate('xpack.cases.getCurrentUser.Error', {
+              defaultMessage: 'Error getting user',
+            }),
+            error: error.body && error.body.message ? new Error(error.body.message) : error,
+            dispatchToaster,
+          });
+          setUser(null);
+        }
+      }
+    };
+    fetchData();
+    return () => {
+      didCancel = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [security]);
```

Comment:
I _think_ you copied this over, but go ahead and remove the eslint-disable here and add `dispatchToaster` to the dependency list. Some amateur (ahem, me) wrote eslint-disable react-hooks/exhaustive-deps on useMemo and useCallback across cases where I should not have. We really should only have that on `useEffect`s

Reply:
Lol, all good. Yea, I took them out. I've just been planning to move everything into the cases plugin first then handle cleanup/refactoring once we had all the files moved over. I don't know any cross component effects some changes will have and figured it would be easier to trace them back if an issue pops up by keeping file changes to as much of a minimum as possible. Probably being overly careful, but yea. I've made these changes though 👍🏾